lib/OpenQA/WebAPI/Controller/API/V1/JobTemplate.pm

```diff
@@ -216,17 +216,33 @@ in the response if any changes to the database were made.
 sub update {
     my $self = shift;
 
+    my $validation = $self->validation;
+    $validation->required('template');
+    $validation->required('schema')->like(qr/^[^.\/]+\.yaml$/);
+    $validation->optional('preview')->num(undef, 1);
+    $validation->optional('expand')->num(undef, 1);
+    $validation->optional('reference');
+    return $self->render(json => {error => ["No $_ specified"]}, status => 400) for @{$validation->failed};
```

Comment:
This loop always returns in the first iteration. It would be better to return a lit of all failed parameters. Rendering this kind of error could be extracted into a generic function/helper.

Reply:
Using a helper would mean changing the type of `error` which is an array, elsewhere it would be a string :thinking:

Reply:
That's true. Maybe we shouldn't be that inconsistent and render a joined string here. And I meant the whole thing should be a helper (from checking the validation and potentially rendering an error).